docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+<!--
+SPDX-FileCopyrightText: 2020 tqtezos
+SPDX-License-Identifier: MIT
+-->
+
+**Overview**
+------------
+
+These specifications were assembled with the following references:
+
+- [*CENTRE Fiat Token design*](https://github.com/centrehq/centre-tokens/blob/78d964a1a8d481ffd8152772d7a66e47df54b3db/doc/tokendesign.md) which is served as a reference specification to tezos stablecoin smart contract.
+
+- Tezos Token Standard:
+  [*FA2*](https://gitlab.com/tzip/tzip/-/blob/76d5f3791bfbfe3c9bf95ad5ec5fc6cbeeca2d0e/proposals/tzip-12/tzip-12.md) and
+  [*ManagedLedger*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-7/ManagedLedger.md)
+
+- [*Michelson Contract Interfaces and Conventions*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-4/tzip-4.md) TZIP which defines `view` and `void` type synonyms
+
+**General Requirements**
+------------------------
+
+- The token(s) must be and FA2 compatible as to facilitate listing on
+  exchanges and interaction with services which support FA2
+
+**Ledger**
+==========
+
+Every address that is stored in ledger is associated with its
+current balance, current allowance and a list of minters.
+
+**Roles**
+=========
+
+The token supports the following list of user roles as is described in
+CENTRE Fiat Token specification and the taxonomy of permission policies that are described in FA2 token specification.
+
+<!-- Here're the roles that are present in current version of stablecoin token.  -->
+
+Below we define some roles for stablecoin contract token.
+
+**owner**
+
+- Can re-assign any role of the token
+
+**masterMinter**
+
+- Can add and remove minters
+
+- Can increase minting allowance for any minter
+
+- Stores the list of current minters within itself, rather
+  than exposing the map of master minter and its minters globally
+  for the whole token
```

Comment:
Yes, I meant that each `masterMinter` has a list of it's sub-minters that can receive tokens from their parent. All minters can depend only on one `masterMinter` preventing their cyclic appearances. Yes, it needs to be clarified that it can increase allowance for any minter within its list.